### PR TITLE
Add test case for NULL NOT DISTINCT composite/partial unique index and expression unique index with CDC changes having conflicts but its table partitioned

### PR DESCRIPTION
**Pull request by priyanshi-yb**
### Describe the changes in this pull request

Adds three live-migration conflict-detection tests, closing coverage for the unique key.

- **Composite NULLS NOT DISTINCT** — `TestLiveMigrationWithCompositeNullsNotDistinctUniqueIndexConflictDetection` covers a multi-column `(part_a, part_b) NULLS NOT DISTINCT` index. It exercises the mixed value/NULL and all-NULL sentinel buckets on the before-before path, which previously existed only in conflict-cache unit tests.
- **NND + partial predicate** — `TestLiveMigrationWithNullsNotDistinctPartialUniqueIndexConflictDetection` covers `(check_id) NULLS NOT DISTINCT WHERE most_recent`. This NND-plus-partial combination had no test at any level. Decoy `most_recent=false` rows with duplicate NULL `check_id` prove the partial predicate is applied.
- **Expression UK under table routing** — `TestLiveMigrationExpressionUniqueIndexCollisionWithTablePartitioning` drives a real `lower(email)` free/reclaim collision with `--cdc-partition-key table`. Existing expression-UK tests only check the guardrail rejection, never a genuine collision. It asserts zero detected conflicts and consistent data, proving table routing suppresses the race structurally.
- Conflict counts assert both a floor (`> 0`) and a delta-derived ceiling (≤ 4000 for the two NND tests), matching the bounded-assertion pattern from the base branch.

### Describe if there are any user-facing changes

No user-facing changes.

### How was this pull request tested?

Three new integration tests under the `failpoint_import` build tag, run against real PostgreSQL and YugabyteDB containers: `go test -tags failpoint_import -run 'TestLiveMigrationWithCompositeNullsNotDistinctUniqueIndexConflictDetection|TestLiveMigrationWithNullsNotDistinctPartialUniqueIndexConflictDetection|TestLiveMigrationExpressionUniqueIndexCollisionWithTablePartitioning' ./yb-voyager/src/testlivemigration/`.

### Does your PR have changes in callhome/yugabyted payloads? If so, is the payload version incremented?

No.

### Does your PR have changes to on-disk structures that can cause upgrade issues?

No.

---

#### On-disk structures potentially causing breaking changes:
| Component        
| :----------------------------------------------: | 
| MetaDB                                           |
| Name registry json                               |
| Data File Descriptor Json                        |
| Export Snapshot Status Json                      |
| Callhome Json                                    |
| Export Status Json                               |
| YugabyteD Tables                                 |
| TargetDB Metadata Tables                         |
| Schema Dump                                      |
| AssessmentDB                                     |
| Migration Assessment Report Json                 |
| Import Data State                                |
| Export and import data queue                     |
| Data .sql files of tables                        |

### CODE REVIEW

File: yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go [hunk 1]
```diff
@@ -3770,3 +3770,427 @@ func TestLiveMigrationCustomCdcPartitionKeyUniqueKeyConflictDetection(t *testing
...
+		},
+	}, 120, 5)
+	testutils.FatalIfError(t, err, "failed to wait for streaming complete")
+
+	require.False(t, lm.GetImportRunner().IsStopped(),
+		"import should keep running during count failpoint mode")
+
+	conflictStats, err := testutils.ReadUniqueKeyConflictStats(uniqueKeyConflictStatsPath)
+	testutils.FatalIfError(t, err, "failed to read unique key conflict stats")
+	require.Greater(t, conflictStats.Total, 0, "composite NND delta should produce UK conflicts")
```
Comment (makalaaneesh) [inline yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go]: [AI][Warning] **Vacuous lower bound on composite-NND conflict count.**

`require.Greater(t, conflictStats.Total, 0, ...)` here (and the per-table variant on the next line) is exactly the vacuous lower bound the standards forbid. `yb-voyager/src/testlivemigration/AGENTS.md` ("Assertions": `>= 0`-style bounds are vacuous) and `yb-voyager/AGENTS.md` ("a vacuous lower bound ... asserts nothing") both require a justified non-trivial lower bound.

This test's delta is structurally identical to `TestLiveMigrationWithUniqueKeyConflictWithNullValuesDetectionCasesNULLSNOTDISTINCT` (line 642), which asserts `> 500` on the same 8-cached-events-per-iteration x 500 workload — so a tight bound is proven achievable.

Failure case: a regression producing only 1 conflict still passes `> 0`, reporting green while NND conflict detection is broken.

Suggested fix:
```go
require.Greater(t, conflictStats.Total, 500, "composite NND delta should produce UK conflicts")
require.Greater(t, conflictStats.ByTable[`"test_schema"."test_live_composite_nnd"`], 500,
    "test_live_composite_nnd should produce UK conflicts")
```
(If the composite case is expected to produce materially fewer conflicts, use whatever tighter bound the workload justifies and state the derivation in the comment — but not 0.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go [hunk 2]
```diff
@@ -3770,3 +3770,427 @@ func TestLiveMigrationCustomCdcPartitionKeyUniqueKeyConflictDetection(t *testing
...
+		},
+	}, 120, 5)
+	testutils.FatalIfError(t, err, "failed to wait for streaming complete")
+
+	require.False(t, lm.GetImportRunner().IsStopped(),
+		"import should keep running during count failpoint mode")
+
+	conflictStats, err := testutils.ReadUniqueKeyConflictStats(uniqueKeyConflictStatsPath)
+	testutils.FatalIfError(t, err, "failed to read unique key conflict stats")
+	require.Greater(t, conflictStats.Total, 0, "NND partial delta should produce UK conflicts")
```
Comment (makalaaneesh) [inline yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go]: [AI][Warning] **Vacuous lower bound on NND-partial conflict count.**

Same issue as the composite-NND case: `require.Greater(t, conflictStats.Total, 0, ...)` here and the per-table variant on the next line under-assert, violating `yb-voyager/src/testlivemigration/AGENTS.md` and `yb-voyager/AGENTS.md` on vacuous lower bounds.

The churned rows are all `most_recent=true`, so the effective workload matches the single-column NND test (`> 500`).

Failure case: a break suppressing NND-partial detection down to a single conflict still passes `> 0`, giving false confidence.

Suggested fix:
```go
require.Greater(t, conflictStats.Total, 500, "NND partial delta should produce UK conflicts")
require.Greater(t, conflictStats.ByTable[`"test_schema"."test_live_nnd_partial"`], 500,
    "test_live_nnd_partial should produce UK conflicts")
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go [hunk 3]
```diff
@@ -3770,3 +3770,427 @@ func TestLiveMigrationCustomCdcPartitionKeyUniqueKeyConflictDetection(t *testing
...
+// conflict-cache unit tests.
+//
+// The unique key is (part_a, part_b) NULLS NOT DISTINCT with part_b held NULL throughout, so
+// every indexed tuple is of the form (X, NULL) or (NULL, NULL) — exactly the mixed value/NULL
+// and all-NULL composite sentinel buckets. part_a is cycled value<->NULL across two PKs (i-1, i)
+// so the same composite tuple is freed on one row and reclaimed on another with a different PK;
+// under partition-by-PK those two events hash to different channels and must be serialized by
+// conflict detection. Because part_b is a constant NULL, uniqueness reduces to part_a under NND,
+// so this delta is valid on the source for the same reason the single-column NND delta is.
+func TestLiveMigrationWithCompositeNullsNotDistinctUniqueIndexConflictDetection(t *testing.T) {
```
Comment (makalaaneesh) [inline yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go:3788]: [AI][Suggestion] **Opaque "gap G5"/"gap G3" references in doc comments.**

The three new doc comments cite "gap G5" (near this function and the NND-partial function) and "gap G3" (near the partitioning function). These reference an external gap-analysis artifact a future reader cannot resolve — the same category as `// fix for #1234` that root `AGENTS.md` ("Comments") says belongs in the PR body, not code. The rest of each doc comment already states the invariant.

Suggested fix: drop the `(gap G5)`/`(gap G3)` parentheticals, or replace with the concrete gap stated in words (e.g. "composite NND was only covered at the unit level") — which the comments already do immediately after, making the token redundant.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by makalaaneesh — COMMENTED** [review]
[AI] Automated review — 3 findings (0 critical, 2 warning, 1 suggestion). See inline comments.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go [hunk 4]
```diff
@@ -3770,3 +3770,427 @@ func TestLiveMigrationCustomCdcPartitionKeyUniqueKeyConflictDetection(t *testing
...
+// expression-unique-index collision (free/reclaim of lower(email) across different PKs) is
+// applied correctly under table routing. The guardrail tests
+// (RejectsPkOnExpressionUniqueIndex / RejectsCustomOnExpressionUniqueIndex) only prove the
+// remedy is enforced and that the table is accepted with non-colliding inserts; none drives a
+// real expression collision. Here the collision is genuine: under partition-by-PK the delete
+// and the reclaiming insert would hash to different channels and race, but with
+// --cdc-partition-key table every users event serializes on one channel, so the target applies
+// them in commit order. The invariant: table routing suppresses the race structurally, so ZERO
+// conflicts are detected and the data stays consistent.
+func TestLiveMigrationExpressionUniqueIndexCollisionWithTablePartitioning(t *testing.T) {
```
Comment (makalaaneesh) [inline yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go]: why "table" specifically? shouldn't we test with "auto" instead?

**Review by makalaaneesh — APPROVED** [review]
LGTM with one question/comment on the expression test case.

File: yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go (hunk 4 above)
Comment (priyanshi-yb) [reply yb-voyager/src/testlivemigration/live_migration_uk_conflict_test.go]: Yeah, auto should also be keeping it in the table, though, but yeah, I can keep it default then in that case.
